pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -144,3 +145,121 @@ func (r *MigPlan) GetRefResources(client k8sclient.Client) (*PlanResources, erro
 
 	return resources, nil
 }
+
+// PV Actions.
+const (
+	PvMoveAction = "move"
+	PvCopyAction = "copy"
+)
+
+// Name - The PV name.
+// StorageClass - The PV storage class name.
+// Action - The PV migration action (move|copy)
+// staged - A PV has been explicitly added/updated.
+type PersistentVolume struct {
+	Name         string `json:"name,omitempty"`
+	StorageClass string `json:"storageClass,omitempty"`
+	Action       string `json:"action,omitempty"`
+	staged       bool
+}
+
+// Update the PV with another.
+func (r *PersistentVolume) Update(pv PersistentVolume) {
+	r.StorageClass = pv.StorageClass
+	r.staged = true
+}
+
+// Collection of PVs
+// List - The collection of PVs.
+// index - List index.
+// --------
+// Example:
+// plan.Spec.BeginPvStaging(false)
+// plan.Spec.AddPv(pvA)
+// plan.Spec.AddPv(pvB)
+// plan.Spec.AddPv(pvC)
+// plan.Spec.EndPvStaging()
+//
+type PersistentVolumes struct {
+	List    []PersistentVolume `json:"persistentVolumes"`
+	index   map[string]int
+	staging bool
+}
+
+// Allocate collections.
+func (r *PersistentVolumes) init() {
+	if r.List == nil {
+		r.List = []PersistentVolume{}
+	}
+	if r.index == nil {
+		r.buildIndex()
+	}
+}
+
+// Build the index.
+func (r *PersistentVolumes) buildIndex() {
+	r.index = map[string]int{}
+	for i, pv := range r.List {
+		r.index[pv.Name] = i
```

Comment:
Looks like @dymurray's idea on using a map got implemented after all! Guess you figured out how to do it without switching the user-provided data type to a map.